test

```diff
@@ -33,6 +33,20 @@ TEST_PKGS=`find . -name \*_test.go | while read a; do dirname $a; done | sort |
 FORMATTABLE=`find . -name \*.go | while read a; do echo $(dirname $a)/"*.go"; done | sort | uniq | egrep -v "$IGNORE_PKGS" | sed "s|\./||g"`
 TESTABLE_AND_FORMATTABLE=`echo "$TEST_PKGS" | egrep -v "$INTEGRATION_PKGS"`
 
+case "$(uname --machine)" in
+	aarch64*)   MACHINE_ARCH="arm64";;
+	arm*)       MACHINE_ARCH="arm";;
+	i?86*)      MACHINE_ARCH="x86";;
+	mips*)      MACHINE_ARCH="mips";;
+	powerpc64*) MACHINE_ARCH="ppc64";;
+	powerpc*)   MACHINE_ARCH="ppc";;
+	x86_64*)    MACHINE_ARCH="amd64";;
+	*)
+		echo "Unknown machine"
+		exit 255
+		;;
+esac
```

Comment:
Is this code block necessary? Listing stuff out is tedious. How about `if [ -z "$GOARCH" ]; then GOARCH=$(go env GOARCH); fi` ? I think treating the GOARCH as the machine arch is fine for the test script.